docs/concepts/data-partitions.md

```diff
@@ -0,0 +1,95 @@
+# Overview of Data Partitions
```

Comment:
Should we link to this documentation from anywhere?

Reply:
We should - but I think that would come in an overhaul to the main doc that links to extended query tags as well.